internal/controllers/synthesis/exec_io.go

```diff
@@ -11,9 +11,44 @@ import (
 	krmv1 "github.com/Azure/eno/pkg/krm/functions/api/v1"
 	"github.com/go-logr/logr"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+func buildPodInput(comp *apiv1.Composition, syn *apiv1.Synthesizer) ([]byte, error) {
+	bindings := map[string]*apiv1.Binding{}
+	for _, b := range comp.Spec.Bindings {
+		b := b
+		bindings[b.Key] = &b
+	}
+	refs := map[string]*apiv1.Ref{}
+	for _, r := range syn.Spec.Refs {
+		r := r
+		refs[r.Key] = &r
+	}
+
+	inputs := []*unstructured.Unstructured{}
+	for key, r := range refs {
+		b, ok := bindings[key]
+		if !ok {
+			return nil, fmt.Errorf("input %q is referenced, but not bound", key)
```

Comment:
Oooh maybe a validating webhook in the future...

Reply:
Yup, I'll add a TODO